### PR TITLE
Clarify and unify naming of thread-related constants (stacksize and priority)

### PR DESCRIPTION
**Pull request by x3ro**
Changes as discussed in #2725 

1. For base stacksizes: drop the `KERNEL_CONF` prefix in favor of `THREAD`.
    * `KERNEL_CONF_STACKSIZE_DEFAULT` becomes `THREAD_STACKSIZE_DEFAULT`. 
1. For supplemental stacksizes: drop the `KERNEL_CONF` prefix in favor of `EXTRA`.
    * `KERNEL_CONF_STACKSIZE_PRINTF` becomes `THREAD_EXTRA_STACKSIZE_PRINTF`.

In addition:

1. Thread priorities previously called `PRIORITY_...` are now prefixed with `THREAD_`
1. `MINIMUM_STACK_SIZE` adapted to the naming scheme of the other stacksizes, thus becoming `THREAD_STACKSIZE_MINIMUM`

### CODE REVIEW

File: boards/qemu-i386/include/cpu-conf.h [hunk 1]
```diff
@@ -25,12 +25,12 @@ extern "C" {
 
 /* FIXME: This file is just a filler. The numbers are entirely random ... */
 
-#define KERNEL_CONF_STACKSIZE_DEFAULT      (8192)
-#define KERNEL_CONF_STACKSIZE_IDLE         (8192)
-#define KERNEL_CONF_STACKSIZE_PRINTF       (8192)
-#define KERNEL_CONF_STACKSIZE_PRINTF_FLOAT (8192)
+#define THREAD_STACKSIZE_DEFAULT      (8192)
+#define THREAD_STACKSIZE_IDLE         (8192)
+#define THREAD_EXTRA_STACKSIZE_PRINTF       (8192)
```
Comment (OlegHahm) [inline boards/qemu-i386/include/cpu-conf.h]: Indentation is off.

**Comment by OlegHahm** [comment]
You're planning to squash? Otherwise the commit messages are missing a proper prefix.

**Comment by x3ro** [comment]
Yep, planning to squash

File: core/include/thread.h [hunk 2]
```diff
@@ -35,14 +35,65 @@
...
+ * @def THREAD_PRIORITY_MIN
+ * @brief Least priority a thread can have
+ */
+#define THREAD_PRIORITY_MIN            (SCHED_PRIO_LEVELS-1)
+
+/**
+ * @def THREAD_PRIORITY_IDLE
+ * @brief Priority of the idle thread
+ */
+#define THREAD_PRIORITY_IDLE           THREAD_PRIORITY_MIN
```
Comment (haukepetersen) [inline core/include/thread.h]: I would prefer to put parenthesis around this

**Comment by haukepetersen** [comment]
ACK when comment above is addressed

File: core/include/thread.h (hunk 2 above)
Comment (x3ro) [inline core/include/thread.h]: Even though it just maps to THREAD_PRIORITY_MIN which is in parentheses?

**Comment by OlegHahm** [comment]
ACK

**Comment by BytesGalore** [comment]
looks good please squash, then lets see what travis says

**Comment by authmillenon** [comment]
Please rebase

**Comment by x3ro** [comment]
@authmillenon Will do. Travis was unhappy about smth and I didn't manage to fix that yet, but I'll give my best to get it merged asap.

**Comment by OlegHahm** [comment]
Any news?

**Comment by x3ro** [comment]
Since I've touched a lot of files in this PR, I also have to deal with [a lot of](https://gist.github.com/x3ro/b419b05779f9f118991a) doxygen warnings that I have no clue how to fix. How do we handle this?

**Comment by OlegHahm** [comment]
Travis should only complain if the PR introduces doxygen warnings.

**Comment by x3ro** [comment]
Does it not complain about all warnings in files that have been changed in a PR? That was my impression so far.

**Comment by OlegHahm** [comment]
See https://github.com/RIOT-OS/RIOT/blob/master/dist/tools/doccheck/check.sh#L12:
```
  ERR_DIFFFILTER="--diff-filter=AC"
 WARN_DIFFFILTER="--diff-filter=MR"
```

Hence, only created and copied files should create errors, the rest should output only warnings.

**Comment by x3ro** [comment]
@OlegHahm The last Travis build passed but there was a minor merge conflict :( Can I merge this as soon as Travis comes back with a green light? It's ACKed otherwise, right?

**Comment by OlegHahm** [comment]
Yes, you can.

**Comment by OlegHahm** [comment]
`tests/driver_kw2xrf` needs an update. :weary:

**Comment by x3ro** [comment]
Build passed [over at my fork's Travis](https://travis-ci.org/x3ro/RIOT/builds/63485526). Is it okay if I merge?

**Comment by OlegHahm** [comment]
Works for me.

**Comment by x3ro** [comment]
Then brace for impact :boom:

**Comment by jremmert-phytec-iot** [comment]
Sry, this is too late now and maybe it´s a dump question. Why are the old auto-init apps introduced again? And why in this PR?
 * eg. ```tests/driver_at86rf2xx/auto_init_ng_netif/netif_app.c```

> tests/driver_kw2xrf needs an update. :weary:

What exactly needs to be updated there?

**Comment by x3ro** [comment]
If they should not be there, then I probably messed up one of the many rebases :(

**Comment by jremmert-phytec-iot** [comment]
Ah ok, yes they should not be there. The PR that changed the auto-init process is #2901. Maybe this was merged before your last rebase or so.

But this doesn´t break anything, so the files could just be deleted by any follow-up or so. Just asked because this could affect the driver implementation for kw2xrf.

**Pull request merged**